src/univariate/continuous/arcsine.jl

```diff
@@ -33,9 +33,13 @@ end
 
 Arcsine(a::T, b::T) where {T<:Real} = Arcsine{T}(a, b)
```

Comment:
While redeclaring everything can we drop the excessiive type constraints?
Let numbers be numbers

Reply:
maybe that should be its own PR.

Reply:
what's the rationale for dropping the type constraint (these are also useful as type signatures)

Reply:
It breaks the the abiility to use weirder number types.
In general `Real` is not a bad one.
Worse is the Arrays, (it is just this occured  first)

where something is decleared as taking an `Array`,
rather than an `AbstractArray`.
Because it breaks GPUs, Flux, and any of cool lazy array types.

Reply:
Probably better do this in another PR, this one is already big enough.

Reply:
Yeah, after I get this done I want to loosen up the array types.

Are there any non-`Real` number types that would be useful? The only one I can think of are Unitful ones, but even those wouldn't really work correctly (e.g. for a gamma distribution, it makes sense for the scale parameter to have a unit, but the shape one should still be unitless).